comp/dogstatsd/server/server_test.go

```diff
@@ -36,9 +37,10 @@ import (
 type serverDeps struct {
 	fx.In
 
-	Server Component
-	Config configComponent.Component
-	Log    log.Component
+	Server        Component
+	Config        configComponent.Component
+	Log           log.Component
+	Demultiplexer demultiplexer.FakeSamplerMock
```

Comment:
Is it weird that we are using custom dependencies struct for testing? The dependencies of the server do not declare the demultiplexer. Would you be able to explain a bit why we need them in the tests?

Reply:
I think it's because the actual server is still using the aggregator package, I guess in the end it should also use the component, but only the tests were changed for this PR

Reply:
That is the part that I need clarification on. Is this PR supposed to make the dogstatsd server use the demultiplexer component? If so it looks like that we are missing that part.

Reply:
> Is it weird that we are using custom dependencies struct for testing?

Can you explain why? If a test needs several components, then it makes sense to have struct with `fx.In` containing several components?

> The dependencies of the server do not declare the demultiplexer

Tests using dogstatsd server need also a demultiplexer component. Having a single `fxutil.Test` is important to share components and so this means having a single  `fx.In` struct . The naming of `serverDeps` can probably be improved (open to suggestion).

> I think it's because the actual server is still using the aggregator package, I guess in the end it should also use the component, but only the tests were changed for this PR

Not really, tests just use both dogstatsd server and a demultiplexer component.

> Is this PR supposed to make the dogstatsd server use the demultiplexer component

No, this PR is about reducing the usage of aggregator package in favor of `comp/aggregator/demultiplexer`. It it not about removing all usages but some of them. It is a preliminary work before moving code from `pkg/aggregator` to `comp`.

Reply:
> Not really, tests just use both dogstatsd server and a demultiplexer component.

That sounds weird to me 😄 I would have imagined that the test for the dogstatsd server only cared about testing the server logic. If the server in production does not use the demultiplexer component, why does the test use it? 

> No, this PR is about reducing the usage of aggregator package in favor of comp/aggregator/demultiplexer. It it not about removing all usages but some of them. It is a preliminary work before moving code from pkg/aggregator to comp.

Thanks for the clarification.

Reply:
> That sounds weird to me 😄 I would have imagined that the test for the dogstatsd server only cared about testing the server logic. If the server in production does not use the demultiplexer component, why does the test use it?

This is because dogstatsd server is not finished and still have a [Start](https://github.com/DataDog/datadog-agent/blob/0a5200071982543cfca6c690fafd34a607ba0a2a/comp/dogstatsd/server/component.go#L23) method that requires a demultiplexer instance.